android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -315,3 +325,48 @@ fun String.extractLogicalIdUuid() = this.substringAfter("/").substringBefore("/"
 fun Resource.addTags(tags: List<Coding>) {
   tags.forEach { this.meta.addTag(it) }
 }
+
+suspend fun Task.updateDependentTaskDueDate(defaultRepository: DefaultRepository): Task {
+  return apply {
+    val isUpcoming = status == Task.TaskStatus.REQUESTED
+    val isOverdue =
+      (status == Task.TaskStatus.READY || status == Task.TaskStatus.INPROGRESS) &&
+        executionPeriod.end.before(Date())
+    val isDue = status == Task.TaskStatus.INPROGRESS
+    if (!(isUpcoming || isOverdue || isDue) || partOf.isEmpty()) {
+      return this
+    }
+    partOf.forEach { task ->
+      val dependentTask = defaultRepository.loadResource(task.reference as Reference) as? Task
```

Comment:
The cast for `task.reference as Reference` here will never succeed reason task.reference is a String.

Reply:
@ellykits dependentTaskOutputValue.valueToString() is not a resource but rather JSON Array of 
```kotlin 
Task.TaskOutputComponent. ie 
               Task.OutputComponent(
                 CodeableConcept(), Type)
               )
```